fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1384,6 +1405,11 @@ private RecordQueryPlan tryToConvertToCoveringPlan(@Nonnull PlanContext context,
         for (KeyExpression resultField : context.query.getRequiredResults()) {
             resultFields.addAll(resultField.normalizeKeyForPositions());
         }
+        if (filterFields != null) {
+            for (KeyExpression filterField : filterFields) {
+                resultFields.addAll(filterField.normalizeKeyForPositions());
```

Comment:
I think that this part here only works because `findFilterCoveredFields()` ignored nested repeated fields entirely. That's fine with me, but I think it needs a comment to that effect or someone might mess this up later.

Reply:
I put the comment on `findFilterCoveredFields`; I think I captured the essence of the tricky cases.